api/src/extensions/lib/get-extensions-settings.ts

```diff
@@ -28,11 +28,44 @@ export const getExtensionsSettings = async ({
 	const existingSettings = await service.extensionsItemService.readByQuery({ limit: -1 });
 
 	const newSettings: ExtensionSettings[] = [];
+	const removedSettingIds: string[] = [];
 
 	const localSettings = existingSettings.filter(({ source }) => source === 'local');
 	const registrySettings = existingSettings.filter(({ source }) => source === 'registry');
 	const moduleSettings = existingSettings.filter(({ source }) => source === 'module');
 
+	const updateBundleEntriesSettings = (
+		extensionSettings: ExtensionSettings,
+		extension: Extension,
+		settings: ExtensionSettings[],
+	) => {
+		if (extension.type !== 'bundle') return;
+
```

Comment:
Let's move this check outside of this function (and expect only `BundleExtension` to be passed)

```suggestion
```